diff --git a/lib/elliptic/ec/index.js b/lib/elliptic/ec/index.js
index a3fe37e3a5f33f1519dbd605c9c8a47a14155deb..06ebd42dd9da5aad6c06ef7826d4c77625470b65 100644
--- a/lib/elliptic/ec/index.js
+++ b/lib/elliptic/ec/index.js
@@ -154,7 +154,7 @@ EC.prototype.sign = function sign(msg, key, enc, options) {
     var k = options.k ?
       options.k(iter) :
       new BN(drbg.generate(this.n.byteLength()));
-    k = this._truncateToN(k, true);
+    k = this._truncateToN(k, true, this.n.bitLength());
     if (k.cmpn(1) <= 0 || k.cmp(ns1) >= 0)
       continue;
 
